src/context-providers/yup/custom-conditions/index.ts

```diff
@@ -41,3 +43,38 @@ YupHelper.addCondition("mixed", "equalsField", (values: unknown[], matches: unkn
 			return isEqual(values, fn.parent[`${matches}`]);
 	}
 });
+
+YupHelper.addCondition(
+	"mixed",
+	"withinDays",
+	(
+		value: string,
+		withinDays: {
+			numberOfDays: number;
+			specificDate?: string;
+			dateFormat?: string;
+		}
+	) => {
+		if (!value) return true;
+		const { numberOfDays, specificDate, dateFormat = "yyyy-MM-dd" } = withinDays;
```

Comment:
```suggestion
		const { numberOfDays, specificDate, dateFormat = "uuuu-MM-dd" } = withinDays;
```

use `uuuu` instead of `yyyy`: https://stackoverflow.com/questions/41177442/uuuu-versus-yyyy-in-datetimeformatter-formatting-pattern-codes-in-java